numba/typeinfer.py

```diff
@@ -1323,8 +1323,20 @@ def typeof_global(self, inst, target, gvar):
                 # as a global variable
                 typ = types.Dispatcher(_temporary_dispatcher_map[gvar.name])
             else:
-                msg = _termcolor.errmsg("Untyped global name '%s':") + " %s"
-                e.patch_message(msg % (gvar.name, e))
+                nm = gvar.name
+                msg = _termcolor.errmsg("Untyped global name '%s':" % nm)
+                msg += " %s" # interps the actual error
+
+                # if the untyped global is a numba internal function then add
+                # to the error message asking if it's been imported.
+                from numba import special
+                numba_internals = special.__all__[1:]
```

Comment:
The `[1:]` is easy to forgot if we change `typeof` or the ordering in `__all__`.  I think we should just include all of `__all__`.

Reply:
The difficulty here is that using `typeof` would be illegal even if in the compilation scope, but then I guess it'd be untypeable and that'd also be fine. I'll go with the suggested.

Reply:
d038d8e fixes